### ISSUE TITLE
ActiveSync: Error codes as messages 

### ISSUE THREAD

**Issue by benbucksch** [mustang-im/mustang#223]
### Actual result

"ActiveSync Sync status 7"

![image](https://github.com/user-attachments/assets/78cfd4a9-7c6f-469d-8cba-dfa8ee74f7b5)

Stack
```
EASError: ActiveSync Sync status 7
    at ActiveSyncEMail.markRead (ActiveSyncEMail.ts:87:13)
```

### Expected result

* A user-understandable error message.
* For the error itself, see #224

### Steps

1. Turn the numeric ActiveSync error codes like `"7"` into an enum
2. In the `ActiveSyncError` class, translate them into user-readable error messages.

**Comment by NeilRashbrook** [comment]
It doesn't help that some error codes are operation-dependent. For instance, for FolderSync, 10 means malformed request, but that same error is code 6 for a Sync operation.

**Comment by benbucksch** [comment]
@NeilRashbrook I see that there's e.g. a ping status and a policy status etc, and we treat them all as EASError.code. Are they actually completely different codes, defined per method? Or is there a common set of codes? Maybe in a certain range? What do you propose?

 E.g. compare `listenForPings()`, which already has a switch based on many of these codes, vs. Provision status.

**Comment by NeilRashbrook** [comment]
Codes over 100 are global, so for instance code 131 would be returned for any command if the user's account is disabled. Lower values are per-command, although usually code 1 is the only success code; the exceptions being Ping, where 1 means the Ping did nothing successfully and 2 means that there are changes (since we also want to handle codes 5 and 6, I handle the Ping status separately) and MoveItems, where there's a secondary status variable which is 3 for success (other secondary status variables exist but they follow the convention where 1 is success).

Codes up to 100 are completely different codes defined per method. (In the case of secondary status variables, they may even use different WBXML code pages, although we just decode them all to the name "Status".)

(WBXML sends integers in decimal so that they're strings when we receive them; while we're supposed to know which tags correspond to numeric values I just run them through `sanitize.integer` where necessary although I don't even bother with that for status codes since we can just compare them to the string `"1"` anyway.)

**Comment by benbucksch** [comment]
I got error code 111 recently

**Comment by benbucksch** [comment]
Fixed in #312 and 2d9fed5c
